lib/cancan/rule.rb

```diff
@@ -125,3 +137,28 @@ def wrap(object)
     end
   end
 end
+
+class SubjectClassMatcher
```

Comment:
Can you explain me why is this needed? If we are already building all the new rules. Please move it into a file itself with some comments about what is it doing and why is necessary

Reply:
It is used to match subclasses and to upmatch from children to their ancestors to look for sti relationships